src/main/scala/com/workday/warp/dsl/WarpMatchers.scala

```diff
@@ -26,23 +29,24 @@ trait WarpMatchers {
     * @return a [[MatchResult]] for whether `responseTime` is less than `threshold`.
     */
   private[this] def matchResult(threshold: Duration,
-                                responseTimes: List[Option[Duration]],
+                                responseTimes: Seq[Duration],
                                 negate: Boolean = false): MatchResult = {
-    // check if there exists a responseTime greater than the threshold; xor for negating the result
-    // responseTimes.flatten removes any None values in List[Option[Duration]] and returns a List[Duration]
-    val failed: Boolean = (responseTimes.flatten exists { _ > threshold }) ^ negate
+    val failed: Boolean = {
+      val exceeded: Boolean = responseTimes.exists(_ > threshold)
+      if (negate) !exceeded else exceeded
```

Comment:
was there a reason to remove the `^ negate` and replace it with an explicit if-else? The if-else seems clunky, but I get that maybe the xor requires more thought...

Reply:
yup, i thought it'd read more straight forward with an explicit if-else